server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java

```diff
@@ -197,9 +197,11 @@ public Task createTask(long id, String type, String action, TaskId parentTaskId,
         } finally {
             if (processed == false) {
                 // something went wrong - unregistering task
-                logger.warn("Persistent task [{}] with id [{}] and allocation id [{}] failed to create", task.getAction(),
-                        task.getPersistentTaskId(), task.getAllocationId());
+                ParameterizedMessage e = new ParameterizedMessage("Persistent task [{}] with id [{}] and allocation id " +
+                    "[{}] failed to create", task.getAction(), task.getPersistentTaskId(), task.getAllocationId());
+                logger.warn(e.getFormattedMessage());
                 taskManager.unregister(task);
+                notifyMasterOfFailedTask(taskInProgress, new RuntimeException(e.getFormattedMessage()));
```

Comment:
we should rather capture the exception thrown and use that here

Reply:
I'll see what I can do. The setup of this area is tricky since it's relying on `processed` flag and a finally clause rather than exception catching.  Was trying to keep changes minimal, but I think I can stash the caught exception and use it in the finally clause.